rbc/remotejit.py

```diff
@@ -371,24 +388,127 @@ def __call__(self, *arguments, **options):
                 f"{type(self.remotejit).__name__}."
             )
             raise UnsupportedError(msg)
+        return CommonNameCallers(self.func.__name__, [self])(*arguments, **options)
 
-        device = options.get('device')
-        targets = self.remotejit.targets
-        if device is None:
-            if len(targets) > 1:
-                raise TypeError(
-                    f'specifying device is required when target has more than'
-                    f' one device. Available devices: {", ".join(targets)}')
-            device = tuple(targets)[0]
-        target_info = targets[device]
-        with target_info:
-            atypes = tuple(map(Type.fromvalue, arguments))
-            ftype = self.signature.best_match(self.func, atypes)
-            key = self.func.__name__, ftype
-            if key not in self._is_compiled:
-                self.remotejit.remote_compile(self.func, ftype, target_info)
-                self._is_compiled.add(key)
-            return self.remotejit.remote_call(self.func, ftype, arguments)
+
+class CommonNameCallers:
+    """A collection of Caller instances holding functions with a common name.
+    """
+    def __init__(self, name, callers):
+        self.name = name
+        assert callers  # at least one caller must be specified
+        self.remotejit = callers[0].remotejit
+        self.callers = callers
+
+    def __repr__(self):
+        return f"{type(self).__name__}({self.name}, {self.callers})"
+
+    def __str__(self):
+        lst = []
+        for caller in self.callers:
+            lst.append(str(caller.signature))
+        sigs = '\n  '.join(lst)
+        return f'{self.name}:\n  {sigs}'
```

Comment:
I think str result should represent the actual content of the `Signature`:
```
str(myincr) -> Signature(['T(T)', T=[...], ...])
str(myincr.normalized()) -> Signature(['int32(int32)', ...])
```
or similar.

The above is not ideal and I think a better approach is needed here because `myincr` is exposed to end-user.

Reply:
Fixed. See `test_udf/udtf_string_repr` in `tests/test_omnisci_caller.py` for updated behavior.